File: example/javascriptlib/basic/1-simple/qux/src/qux.ts
import {User} from "foo/foo";
import {defaultRoles} from "foo/bar/bar";

/**
 * Generate a user object based on command-line arguments
 * @param args Command-line arguments
 * @returns User object
 */
export function generateUser(args: string[]): User {
    return {
        firstName: args[0] || "unknown", // Default to "unknown" if first-name not found
        lastName: args[1] || "unknown", // Default to "unknown" if last-name not found
        role: defaultRoles.get(args[2], ""), // Default to empty string if role not found
    };
}

// Main CLI logic
if (process.env.NODE_ENV !== "test") {
    const args = process.argv.slice(2); // Skip 'node' and script name
    const user = generateUser(args);

    console.log(defaultRoles.toObject());
    console.log(args[2]);
    console.log(defaultRoles.get(args[2]));
    console.log("Hello " + user.firstName + " " + user.lastName + " " + user.role);
}

File: example/extending/python/3-python-module-deps/qux/src/qux.py
from foo.bar.src.bar import add # type: ignore
from foo.src.foo import multiply # type: ignore
import sys
def divide(a: int, b: int) -> float: return a/b
def main() -> None:
    x = int(sys.argv[1])
    y = int(sys.argv[2])
    print(f"Add: {x} + {y} = {add(x, y)} | Multiply: {x} * {y} = {multiply(x, y)} | Divide: {x} / {y} = {divide(x, y)}")
if __name__ == "__main__":
    main()